src/spawn.c

```diff
@@ -577,7 +604,83 @@ static gboolean spawn_async_with_pipes(const gchar *working_directory, const gch
 
 	spawned = g_spawn_async_with_pipes(working_directory, full_argv, envp,
 		G_SPAWN_SEARCH_PATH | (child_pid ? G_SPAWN_DO_NOT_REAP_CHILD : 0), NULL, NULL,
-		child_pid, stdin_fd, stdout_fd, stderr_fd, error);
+		child_pid, stdin_fd, stdout_fd, stderr_fd, &gerror);
+
+	if (!spawned)
+	{
+		gint en = 0;
+		const gchar *message = gerror->message;
+
+		/* try to cut glib citing of the program name or working directory: they may be long,
+		   and only the caller knows whether they're UTF-8. We lose the exact chdir error. */
+		switch (gerror->code)
```

Comment:
About convoluted logic, well - after replacing the *entire* spawning under Windows, reverting a G_SPAWN error code back to to errno is only a small touch. Why should I regard g_spawn under *nix as a sacred cow?.. :)